lib/action/haptic_action.dart

```diff
@@ -17,6 +18,9 @@ class HapticAction extends EnsembleAction {
 
   final HapticTypes type;
 
+  factory HapticAction.fromMap(dynamic inputs) =>
```

Comment:
combine fromYaml() with this function. takes in dynamic and check if it's a simple `Map` which can be object or yaml.